### PR TITLE
Adding the install_machine_driver_certificate action method

### PR DESCRIPTION
**Pull request by bish22ah**
Installs a driver certificate, (.cer file), on the specified machine,
example usage:
                                                     (.cer file path)
  tools.install_machine_driver_certificate('CL1', './VirtIOTestCert.cer')

Signed-off-by: Bishara AbuHattoum <bishara@daynix.com>

### CODE REVIEW

File: lib/rtoolsHCK.rb [hunk 1]
```diff
@@ -1182,6 +1182,25 @@ def do_install_machine_driver_package(machine,
...
+    end
+    r_cer
+  end
+
+  def do_install_machine_driver_certificate(machine, l_cer)
+    r_cer = do_upload_driver_certificate_files(machine, l_cer)
+    windows_path = r_cer.tr('/', '\\')
+    command1 = "certmgr.exe -add #{windows_path} -s -r localMachine root"
+    command2 = "certmgr.exe -add #{windows_path} -s -r localMachine trustedpub"\
+               'lisher'
```
Comment (20lives) [inline lib/rtoolsHCK.rb]: Cutting words in the middle can be easily avoided (and also not duplicate code).
I would suggest something like that maybe:
```
  def do_install_machine_driver_certificate(machine, l_cer)
    r_cer = do_upload_driver_certificate_files(machine, l_cer)
    windows_path = r_cer.tr('/', '\\')
    add_cert_to_store(machine, 'root', windows_path)
    add_cert_to_store(machine, 'trustedpublisher', windows_path)
  end

  def add_cert_to_store(machine, store, cer_path)
    command = "certmgr.exe -add #{cer_path} -s -r localMachine #{store}"
    machine_run(machine, command)
  end
```

**Review by 20lives — CHANGES_REQUESTED** [review]

**Review by 20lives — APPROVED** [review]

File: lib/rtoolsHCK.rb (hunk 1 above)
Comment (bish22ah) [reply lib/rtoolsHCK.rb]: What about now, I took your advice and pushed the changes.

**Review by 20lives — CHANGES_REQUESTED** [review]
The command should be:
`certutil -enterprise -f -v -AddStore <store> <certificate>`

File: lib/rtoolsHCK.rb [hunk 2]
```diff
@@ -1211,6 +1233,23 @@ def install_machine_driver_package(machine,
     end
   end
 
+  # == Description
+  #
+  # Installs a driver certificate, (.cer file), on a machine.
```
Comment (YanVugenfirer) [inline lib/rtoolsHCK.rb:1238]: does it work with pfx as well?

**Review by YanVugenfirer — COMMENTED** [review]
Check my comment regarding pfx

File: lib/rtoolsHCK.rb (hunk 2 above)
Comment (bish22ah) [reply lib/rtoolsHCK.rb:1238]: No.
The script that installs the certificate ignores the pfx file:
https://github.com/virtio-win/kvm-guest-drivers-windows/blob/master/Tools/InstallCertificate.bat
I assumed it was not essential.
I can add the support easily.

**Comment by bish22ah** [comment]
@20lives the script begs to differ:
https://github.com/virtio-win/kvm-guest-drivers-windows/blob/master/Tools/InstallCertificate.bat

**Comment by YanVugenfirer** [comment]
The certificate might be pfx, please support it

**Comment by bish22ah** [comment]
Closing, sending a new PR with the new approach of exporting the certificate from the driver file and installing it.

**Pull request closed without merging**